packages/core/src/events/input/promptTemplate/base.ts

```diff
@@ -1,43 +1,49 @@
 import { Callable, CallableConfig } from "../../../record/callable";
-import { baseValidator } from "../../../utils/pValidators/baseValidator";
+import { variableValidator } from "../../../utils/promptTemplateValidator/variableValidator";
 import { BaseEvent, BaseEventParams } from "../../base";
 export interface PromptTemplateParams extends BaseEventParams {
     /**
      * Clearly declare the input variables inside the template
      */
-    inputVariables?: string[];
+    inputVariables?: string[] | undefined;
 
     /**
      * the template string
      */
     template?: string | "";
 
-    validators?: baseValidator | undefined;
+    validator?: variableValidator | undefined;
+
+    partialVariables?: string[] | undefined;
 }
 
-export abstract class basePromptTemplate <
-    CallInput = PromptTemplateParams,
-    CallOutput = string[],
+export class basePromptTemplate <
+    CallInput = Record<string,any>,
+    CallOutput = string,
     CallOptions extends CallableConfig = CallableConfig,
   >
   extends BaseEvent<CallInput, CallOutput, CallOptions>
   implements PromptTemplateParams
   {
+    _namespace: string[];
```

Comment:
you should not init this as undefined in a non-abstract Callable class. please check [ContextSplitter](https://github.com/VictorS67/encre/blob/5c3e8e823bca4d3d974695e065e7fee6d599d28c/packages/core/src/events/input/transform/splitter.ts#L29) to see how to assign this value.